### ISSUE TITLE
[Feature] Manage DLC and patches folder

### ISSUE THREAD

**Issue by XargonWan** [rommapp/romm#228]
My roms folder, expacially for the newer systems like 3DS, Vita and PS3 are including `patches` and `dlc` subfolders.
Romm now is trying to match these folder with IGDB producing some unwanted game entries.

Example structure:
```
roms/ps3
- dlc
-- game1/dlc1
-- game1/dlc2
- patches
-- game1
-- game2
- game1.ps3
- game2.ps3
```

**Describe the solution you'd like**
Romm will recognize `dlc` and `patches` folder (name is customizable via config.yml) inside each system folder and will not scarape them.
Moreover it could put a fancy grahpics instead of IGDB logo.
Clicking on the folder it will load a screen that may be different from the game page as dlc and patches may not be scraped (we need to think about how to organize it actually).

**Comment by XargonWan** [comment]
Discussed even here: https://github.com/zurdi15/romm/issues/404#issuecomment-1793548282

**Comment by miltuss** [comment]
This feature will be extremely useful, I realize it is in high demand!

**Comment by RoiArthurB** [comment]
I'd love to see such a feature implemented as well, however I would more follow this kind of structure : 
```
roms/ps4--1
Game1
├── DLC
│   ├── DLC1.pkg
│   ├── DLC2.pkg
│   ├── DLC3.pkg
├── Game1.pkg
└── Patch
   ├── Patch1.pkg
    └── Patch2.pkg
Game2
├── Game2.pkg
└── Patch
   ├── Patch1.pkg
    └── Patch2.pkg
```

This allow to less repeat game's folder name, limit typing mistake and probably ease the coding structure (from the game folder, have both DLC and Patch if folders are available; instead of searching for a name in another folder).

**Comment by miltuss** [comment]
> This allow to less repeat game's folder name, limit typing mistake and probably ease the coding structure (from the game folder, have both DLC and Patch if folders are available; instead of searching for a name in another folder).

I completely agree with you. Organizing all the files of a game in the same folder is the best way to do it, which is why I made a feature request https://github.com/rommapp/romm/issues/1189 for that Romm automatically organizes games in folders specific to each game.

**Comment by Fifthdread** [comment]
Can we get this added? I ask because I started using RomM to manage my local GoG install files. When I download with my tool, it creates a folder for the game itself (currently works as expected) and within that folder, a dlc, extras, and patches folder.

It'd be nice to see even super basic subfolder support, so I could download content via RomM.

I currently use a tool called lgogdownloader to download all my GoG content. Running this tool regularly ensures that I have all the latest files at all times, but if I have to move files around every time I run an update, that'll grow old quick. lol

Thanks

**Comment by gantoine** [comment]
@Fifthdread I'm working on it now, should be out in a couple releases from now.

**Comment by gantoine** [comment]
Basic folder support is available in the latest beta (`3.8.0-beta.2`) and slated for release shortly! The updated supported folder structure can be found here: https://github.com/rommapp/romm/tree/master?tab=readme-ov-file#folder-structure
